src/Mvc/test/WebSites/RoutingWebSite/StartupForDynamic.cs

```diff
@@ -61,7 +61,7 @@ public override ValueTask<RouteValueDictionary> TransformAsync(HttpContext httpC
                 var results = new RouteValueDictionary();
                 foreach (var kvp in kvps)
                 {
-                    var split = kvp.Split("=");
+                    var split = kvp.Replace("%2F", "/", StringComparison.OrdinalIgnoreCase).Split("=");
```

Comment:
@pranavkm can you take a look at these tests? They were written depending on a bug in the TestServer and not verified with a real server.

Reply:
@pranavkm , do you have any suggestion?